driverapi/driverapi.go

```diff
@@ -91,6 +91,9 @@ type JoinInfo interface {
 	// AddStaticRoute adds a routes to the sandbox.
 	// It may be used in addtion to or instead of a default gateway (as above).
 	AddStaticRoute(destination *net.IPNet, routeType int, nextHop net.IP) error
+
+	// DisableGatewayService tells libnetwork not to provide Default GW for the container
+	DisableGatewayService(bool)
```

Comment:
I don't think the bool parameter is needed. The function name is very explicit already.

The default behavior has to remain the same (which is libnetwork default gateway network support is on), in other words bridge and overlay code does not need to be changed in this PR.